src/app/_components/calendar-view.tsx

```diff
@@ -0,0 +1,468 @@
+      setIsDayView(isMobile);
+    }
+  }, [isMobile, setIsDayView]);
+
+  // Update calendar date when currentDate prop changes
+  useEffect(() => {
+    if (currentDate && calendarRef.current) {
+      const calendarApi = calendarRef.current.getApi();
+      // defer the change date to avoid flushSync error
+      setTimeout(() => {
+        calendarApi.gotoDate(currentDate);
+      }, 0);
+    }
+  }, [currentDate]);
+
+  // Update calendar view when screen size changes
+  useEffect(() => {
+    if (calendarRef.current) {
+      const calendarApi = calendarRef.current.getApi();
+      const currentView = calendarApi.view.type;
+      const targetView = isMobile ? "timeGridDay" : "timeGridWeek";
+
+      // Only change view if it's different from current view
+      if (currentView !== targetView) {
+        // defer the change view to avoid flushSync error
+        setTimeout(() => {
+          calendarApi.changeView(targetView);
+        }, 0);
+      }
+    }
+    // Notify parent of view change
+    if (setIsDayView) {
+      setIsDayView(isMobile);
+    }
+  }, [isMobile, setIsDayView]);
+
+  // Handle event click
+  const handleEventClick = (clickInfo: EventClickArg) => {
+    const event = clickInfo.event;
+    const extendedProps = event.extendedProps;
+
+    // Format the event details for display
+    const startTime = event.start
+      ? new Date(event.start).toLocaleTimeString("en-US", {
+          hour: "numeric",
+          minute: "2-digit",
+          hour12: true,
+        })
+      : "N/A";
+    const endTime = event.end
+      ? new Date(event.end).toLocaleTimeString("en-US", {
+          hour: "numeric",
+          minute: "2-digit",
+          hour12: true,
+        })
+      : "N/A";
+
+    const alertMessage = [
+      `Title: ${event.title}`,
+      `Time: ${startTime} - ${endTime}`,
+      extendedProps.pickupLocation ? `Pickup: ${extendedProps.pickupLocation}` : "",
+      extendedProps.dropoffLocation ? `Dropoff: ${extendedProps.dropoffLocation}` : "",
+      extendedProps.status ? `Status: ${extendedProps.status}` : "",
+      extendedProps.driverId ? `Driver ID: ${extendedProps.driverId}` : "",
+      extendedProps.passengerInfo ? `Passenger: ${extendedProps.passengerInfo}` : "",
+      extendedProps.purpose ? `Purpose: ${extendedProps.purpose}` : "",
+    ]
+      .filter(Boolean)
+      .join("\n");
+
+    alert(alertMessage);
+  };
+
+  // Custom event content renderer
+  const renderEventContent = (eventInfo: EventContentArg) => {
+    const event = eventInfo.event;
+    const extendedProps = event.extendedProps;
+    const status = extendedProps.status;
+
+    // Status icons
+    const getStatusIcon = () => {
+      switch (status) {
+        case "completed":
+          return <Check width="12px" height="12px" />;
+        case "incomplete":
+          return <Cross width="12px" height="12px" />;
+        case "in-progress":
+          return null;
+        default:
+          return null;
+      }
+    };
```

Comment:
What I mean here is that status is of type `any`. We want to make sure that it has a type:

<img width="517" height="86" alt="Image" src="https://github.com/user-attachments/assets/0fcffe7a-aed8-4519-a7fc-e2e8fdcabeff" />

```suggestion
+    const statusIcon =
+      status === bookingStatus.COMPLETED ? <Check width="12px" height="12px" /> :
+      status === bookingStatus.IN_PROGRESS ? <Cross width="12px" height="12px" /> :
+      null;
```